fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSTransaction.java

```diff
@@ -0,0 +1,1299 @@
+        private final String currentLocation;
+        private final String partitionName;
+        private final List<String> fileNames;
+        private final HivePartitionStatistics statisticsUpdate;
+
+        public PartitionAndMore(
+                HivePartition partition,
+                String currentLocation,
+                String partitionName,
+                List<String> fileNames,
+                HivePartitionStatistics statisticsUpdate) {
+            this.partition = Objects.requireNonNull(partition, "partition is null");
+            this.currentLocation = Objects.requireNonNull(currentLocation, "currentLocation is null");
+            this.partitionName = Objects.requireNonNull(partitionName, "partition is null");
+            this.fileNames = Objects.requireNonNull(fileNames, "fileNames is null");
+            this.statisticsUpdate = Objects.requireNonNull(statisticsUpdate, "statisticsUpdate is null");
+        }
+
+        public HivePartition getPartition() {
+            return partition;
+        }
+
+        public String getCurrentLocation() {
+            return currentLocation;
+        }
+
+        public String getPartitionName() {
+            return partitionName;
+        }
+
+        public List<String> getFileNames() {
+            return fileNames;
+        }
+
+        public HivePartitionStatistics getStatisticsUpdate() {
+            return statisticsUpdate;
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                .add("partition", partition)
+                .add("currentLocation", currentLocation)
+                .add("fileNames", fileNames)
+                .toString();
+        }
+    }
+
+    private enum ActionType {
+        DROP,
+        DROP_PRESERVE_DATA,
+        ADD,
+        ALTER,
+        INSERT_EXISTING,
+        MERGE,
+    }
+
+    public static class Action<T> {
+        private final ActionType type;
+        private final T data;
+
+        public Action(ActionType type, T data) {
+            this.type = Objects.requireNonNull(type, "type is null");
+            if (type == ActionType.DROP || type == ActionType.DROP_PRESERVE_DATA) {
+                Preconditions.checkArgument(data == null, "data is not null");
+            } else {
+                Objects.requireNonNull(data, "data is null");
+            }
+            this.data = data;
+        }
+
+        public ActionType getType() {
+            return type;
+        }
+
+        public T getData() {
+            Preconditions.checkState(type != ActionType.DROP);
+            return data;
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                .add("type", type)
+                .add("data", data)
+                .toString();
+        }
+    }
+
+    public synchronized Table getTable(String databaseName, String tableName) {
```

Comment:
Add comment to explain the difference between `return null` and `throw exception`